larpandoracontent/LArThreeDReco/LArTrackFragments/ClearTrackFragmentsTool.cc

```diff
@@ -280,31 +257,47 @@ void ClearTrackFragmentsTool::ProcessTensorElement(ThreeDTrackFragmentsAlgorithm
 
         this->Recluster(pAlgorithm, pCluster, daughterHits, separateHits, deletedClusters, badClusters, pFragmentCluster);
 
-        if (badClusters.count(pFragmentCluster))
+        // ATTN Fragment cluster will be used to build particle, so it shouldn't ever be bad, or deleted
+        if (badClusters.count(pFragmentCluster) || deletedClusters.count(pFragmentCluster))
             throw StatusCodeException(STATUS_CODE_FAILURE);
```

Comment:
This check makes sense.  If this exception gets thrown, do you know where it is caught and what the affect on the remaining reconstruction is?  Does it bail out completely or just skip the remaining logic in this algorithm and how does this impact the tensor if it's thrown?

Reply:
I think this will terminate the algorithm, i.e. exit the tool, exit the algorithm, so it's all over and the tensor is destroyed. I searched for this failure during my uB tests and it didn't occur during the tests with the code as is (but was triggered during earlier tests involved with these updates).